src/compiler.cpp

```diff
@@ -0,0 +1,152 @@
+            // The calls need to be well-ordered
+            os << "{ int32_t tmp = lws_pop(stack); lws_push(stack, lws_pop(stack) - tmp); }";
+            return;
+        case op::MUL:
+            os << "lws_push(stack, lws_pop(stack) * lws_pop(stack));";
+            return;
+        case op::DIV:
+            os << "{ int32_t tmp = lws_pop(stack); lws_push(stack, lws_pop(stack) / tmp); }";
+            return;
+        case op::MOD:
+            os << "{ int32_t tmp = lws_pop(stack); lws_push(stack, lws_pop(stack) % tmp); }";
+            return;
+        case op::PSC: {
+            os << "lws_push(stack,";
+            const auto value = i.m_arg.number;
+            os << static_cast<int32_t>(value) << ");";
+            return;
+        }
+        case op::PSI: {
+            os << "lws_push(stack,";
+            const auto value = i.m_arg.number;
+            os << static_cast<int32_t>(value - (int24_t)'0') << ");";
+            return;
+        }
+        case op::POP:
+            os << "(void)lws_pop(stack);";
+            return;
+        case op::INC:
+            os << "lws_push(stack, (lws_pop(stack) + 1) << 8 >> 8);";
+            return;
+        case op::DEC:
+            os << "lws_push(stack, (lws_pop(stack) - 1) << 8 >> 8);";
+            return;
+        case op::AND:
+            os << "lws_push(stack, lws_pop(stack) & lws_pop(stack));";
+            return;
+        case op::IOR:
+            os << "lws_push(stack, lws_pop(stack) | lws_pop(stack));";
+            return;
+        case op::XOR:
+            os << "lws_push(stack, lws_pop(stack) ^ lws_pop(stack));";
+            return;
+        case op::NOT:
+            os << "lws_push(stack, ~lws_pop(stack));";
+            return;
+        case op::GTC:
+            // FIXME: reports (some?) high unicode characters as WEOF (on some systems?)
+            os << "{ wint_t temp = getwchar(); lws_push(stack, temp == WEOF ? -1 : temp); }";
+            return;
+        case op::PTC:
+            os << "putwchar(lws_top(stack));";
+            return;
+        case op::GTI:
+            os << "{"
+                  "int32_t i = -1;"
+                  "while (!(feof(stdin) || scanf(\"%\" SCNi32, &i))) (void)getchar();"
+                  "lws_push(stack, i);"
+                  "}";
+            return;
+        case op::PTI:
+            os << R"( printf("%" PRId32 "\n", lws_top(stack)); )";
+            return;
+        case op::IDX:
+            os << "lws_push(stack, lws_index(stack, lws_pop(stack)));";
+            return;
+        case op::DUP:
+            os << "lws_push(stack, lws_top(stack));";
+            return;
+        case op::DP2: {
+            const char* str = "lws_push(stack, lws_index(stack, 1));";
+            os << str << str;
+            return;
+        }
+        case op::RND:
+        case op::GDT:
+        case op::GTM:
+            cerr << "Nondeterministic instructions are not yet supported for compiled programs." << endl;
```

Comment:
This should be doable with `time()` and `rand()`, but seeding the RNG is a problem

Reply:
`time()` doesn't provide sub-second precision however, so it might work for `GDT` but not `GTM`.